goal_src/engine/math/transformq.gc

```diff
@@ -5,3 +5,536 @@
+        )
+   (set! (-> s2-0 x) (sin arg0))
+   (set! (-> s2-0 y) 0.0)
+   (set! (-> s2-0 z) (cos arg0))
+   (set! (-> s2-0 w) 1.0)
+   (s3-0 obj s2-0 arg1 arg2)
+   )
+  )
+
+;; definition for method 14 of type trsqv
+(defmethod TODO-RENAME-14 trsqv ((obj trsqv) (arg0 float))
+  (let ((s5-0 (method-of-object obj TODO-RENAME-19))
+        (s4-0 (new 'stack-no-clear 'quaternion))
+        )
+   (set! (-> s4-0 x) (sin arg0))
+   (set! (-> s4-0 y) 0.0)
+   (set! (-> s4-0 z) (cos arg0))
+   (set! (-> s4-0 w) 1.0)
+   (s5-0 obj (the-as vector s4-0))
+   )
+  )
+
+;; definition for method 15 of type trsqv
+(defmethod TODO-RENAME-15 trsqv ((obj trsqv) (arg0 float))
+  (TODO-RENAME-VECTOR-DOT-16 obj arg0)
+  )
+
+;; definition for method 16 of type trsqv
+(defmethod TODO-RENAME-VECTOR-DOT-16 trsqv ((obj trsqv) (arg0 float))
+  (rlet ((acc :class vf)
+         (vf1 :class vf)
+         (vf2 :class vf)
+         (vf3 :class vf)
+         )
+   (let* ((s5-0 (get-quaternion obj))
+          (s1-0 (-> *standard-dynamics* gravity-normal))
+          (s3-0 (quaternion->matrix (new 'stack-no-clear 'matrix) s5-0))
+          )
+    (let ((s4-0 (&-> s3-0 data 8)))
+     (vector-normalize!
+      (vector-flatten!
+       (the-as vector (&-> s3-0 data 4))
+       s1-0
+       (the-as vector s4-0)
+       )
+      1.0
+      )
+     (let ((a0-4 (-> s3-0 data)))
+      (let ((v1-3 (&-> s3-0 data 4)))
+       (.lvf vf1 v1-3)
+       )
+      (.lvf vf2 s4-0)
+      (.outer.product.vf vf3 vf1 vf2)
+      (.svf (the-as (pointer uint128) (&-> a0-4 0)) vf3)
+      )
+     )
+    (let ((a1-5 (matrix-rotate-z! (new 'stack-no-clear 'matrix) arg0)))
+     (matrix*! s3-0 a1-5 s3-0)
+     )
+    (matrix->quaternion s5-0 s3-0)
+    )
+   )
+  )
+
+;; definition for method 25 of type trsqv
+;; WARN: Unsupported inline assembly instruction kind - [mula.s f0, f3]
+;; WARN: Unsupported inline assembly instruction kind - [madda.s f1, f4]
+;; WARN: Unsupported inline assembly instruction kind - [madd.s f0, f2, f5]
+;; WARN: Unsupported inline assembly instruction kind - [mula.s f1, f4]
+;; WARN: Unsupported inline assembly instruction kind - [madda.s f2, f5]
+;; WARN: Unsupported inline assembly instruction kind - [madd.s f1, f3, f6]
+(defmethod TODO-VECTOR-MATH-ISSUES-25 trsqv ((obj trsqv))
+  (local-vars (f0-1 float) (f1-2 float))
+  (rlet ((acc :class vf)
+         (vf1 :class vf)
+         (vf2 :class vf)
+         (vf3 :class vf)
+         )
+   (let* ((s5-0 (get-quaternion obj))
+          (gp-0 (vector-z-quaternion! (new 'stack-no-clear 'vector) s5-0))
+          (s5-1 (vector-y-quaternion! (new 'stack-no-clear 'vector) s5-0))
+          (a1-2 (-> *standard-dynamics* gravity-normal))
+          (v1-2
+           (vector-normalize!
+            (vector-flatten! (new 'stack-no-clear 'vector) a1-2 gp-0)
+            1.0
+            )
+           )
+          )
+    (vector-dot v1-2 s5-1) ; added manually, function was inlined.
+    (let ((f0-2 f0-1))
```

Comment:
I think one of these was set by `vector-dot`.  Right now this uses the `f0-1` and `f1-2` local vars without ever setting them to a value.